### DISCUSSION TITLE
Small blank windows during launch and opening dialogs

### DISCUSSION THREAD

**Discussion by advice2020** [doublecmd/doublecmd#1503] [General]
Double Commander
Version: 1.2.0 alpha
Revision: 11991
Commit: 34a54ef00
Build date: 2025/04/06
Lazarus: 3.6.0.0 (Also noticed this with 3.0.0)
FPC: 3.2.2
Platform: x86_64-Linux-qt5 (also noticed in QT6)
OS version: Debian based Linux distro / GTK based DE
Widgetset library: Qt 5.15.8 (also noticed in QT6.4.2)
Using SSD (Not NVMe drive)

Note:
I am in the process of updating/reopening this post with new information but will leave some of it for now, just not sure if the popups are exactly the same as before.
I had originally thought this might be due to something I was doing wrong when compiling, but after reading something recently this might not be true (can see my post below).
This also occurs when entering an items `Properties` dialog.

-------------------------------------------------------------------------------

So far the compiled program launches / works fine but I notice when launching the program I am getting blank boxes appearing before the main DC window loads which I do not experience when using the pre-build DC QT5 snapshot version.

I have mentioned in a different "Discussion" post that ALL QT5/6 based versions of DC when run in my GTK based DE (for some reason in my distro only),  load with a bit of a pause, takes about 5-6 seconds to load, so I have a little time before the main DC window opens, that maybe others might not.

**If DC splash screen turned off**
-I get a blank 4"x4" window that flashes very quickly in the center of the screen
-Then a blank 2"x2" window that shows up in the upper left cornder of screen
-Then 2x2 window goes away and the DC window loads

**If DC splash screen turned on**
Before launching make DC window 50% width of your screen
-I get a blank 4"x4" window that flashes very quickly in the center of the screen
-Then a blank 2"x2" window that shows up in the upper left cornder of screen
-Then 2x2 window goes away and the DC window loads
-And with DC being 50% of screen you can see that the DC splash screen ends up loading behind DC main window

Everything works fine as far as I know, but just doesn't really look that great when this happens.
If anyone can confirm this happening with a similar setup or if anyone has any ideas of how to make these blank windows not show up it would be greatly appreciated.  Just not sure if it is something I did wrong when compiling?

Thank You

**Comment by advice2020** [comment]
@j2969719
You mentioned that you experience something similar when opening up `Properties` dialog for items, seeing a black box in upper left corner of screen (while using QT6.8.1)
Not sure what causes this, but I do not think it is specific to QT6 because I experience this in QT5.15.8 with `Properties` dialog as well.
This is similar to what I mention in this thread, I experience box in upper left corner at DC launch.
For me both Properties & Launch do not show up as a black box but a very quick box shaped window in upper left corner, that shows header/title bar at top.
I am using an SSD so maybe your system is faster and that is why only shows as a black box, if these issues are the same?

Originally I do not remember experiencing this with any of the officially released versions of DC (which use QT5 as well), I had always thought that it might have been something I was doing incorrectly when compiling, for that reason.  However, I recently tried some versions of the alpha snapshot QT5 Appimages again and now this dialog appears at Launch and Propeties dialogs as well.
I was going to say that it is probably not FPC or Lazarus because I thought DC AppImages were made with older versions of them but just realized this is not true.
AppImage I just tried is using same version of FPC that I am (3.2.2) and even newer version of Lazarus then I am (3.6.0), so this could also be from an update in one of these applications?
```
Version: DC 1.2.0 alpha AppImage snapshot
Revision: 11938
Commit: 8fd3f87
Build date: 2025/03/12
Lazarus: 3.8.0.0
FPC: 3.2.2
Platform: x86_64-Linux-qt5
Widgetset library: Qt 5.15.8
```

Wonder if you are experiencing same issue as I am?
Do you by any change experience same black box in upper left corner when you launch DC?

If you find anything about this, please mention it here, still curious about this.

Thank You

**Comment by alexx2000** [comment]
Looks like a Lazarus bug. Depends on the DE.

**Comment by advice2020** [comment]
Thank you Alex

So it is Lazarus bug that depends on the DE being used.
This one has been around for a while, I first noticed this I think in 3.0.0 and still occurs in 3.8.0
Not sure what to search for, does anyone know if the developers are aware of this?

**Comment by advice2020** [comment]
I forgot about this, this also occurs for me when using `MPV`.
My MPV is set to open in "fullscreen"
When video is loaded/MPV is launched, a small black box appears in upper left corner and then flashes/kind of animates before opening in fullscreen.
Did not like the look of this either, was trying to fix this in the past but never found any answers for this.
Wonder if these two issues are in any way related?
If they are then it would not be Lazarus, because I do not think MPV uses this at all.
So if these issues are the same, maybe it is just window manager specific?

@j2969719 
You mentioned you experienced this with DC, I was wondering if you or any other users, experience this with MPV as well?

I will try to ask around, but if anyone has anything to share about this, please let me know
Thank You

**Comment by j2969719** [comment]
this could be both a window manager artifact and a change in lazarus that made it noticeable. found some old build from 2021 compiled with lazarus 2.1 in my backups and there are no such artifacts visible there.

**Comment by advice2020** [comment]
@j2969719 
Thank you, yeah I mentioned that as well, I remember using snapshots compiled by Alex in the past and never noticing these artifacts.
Then when I started to have to compile using newer Lazarus versions (because of a different Lazarus error), that is when I began noticing these issues.  I originally thought it was because Alex was just better at compiling software then me, but I do not know why I doubted myself, I am way better at compiling then that guy...... :)
Do you by any change experience what I mentioned in `mpv`?
I wish I had older versions of `mpv` to compare to.
I cannot remember if this happened in older versions, I believe I am using version 0.39 (there is newer version 0.40 but I do not have access to it)
If we both experience these artifacts in DC and MPV then it might help to narrow down if Lazarus is part of the issue, because MPV does not have any connection with Lazarus that I know of.
If these artifacts in DC and MPV are part of the same issue of course, they might not have anything to do with each other, but it is strange that they both start up in that upper left corner of screen, 0,0
Anyways, thank you again

If any other users experience this in DC and or MPV please share
Thank You

**Reply by j2969719** [reply]
>Do you by any change experience what I mentioned in mpv?

maybe? cant say for sure, i just dont have good eyesight and fast reaction. but in case of the properties window, even i can register the flickering

in case of icewm without compositor the flickering of the properties window is also noticeable, but after running picom i dont notice anything

**Comment by advice2020** [comment]
(Again NVMe users may not notice this, I am using SSD)
Not sure if this helps figure this out but I downloaded both snapshot AppImage versions

`doublecmd-gtk-latest-x86_64.AppImage`
`doublecmd-qt-latest-x86_64.AppImage`

Revision: 12195
Commit: 0ead73a7
Build date: 2025/05/18
Lazarus: 3.8.0.0
FPC: 3.2.2

Both are using the same Lazarus version which is even newer then the one I have access to (which is 3.6)
Same FPC version as I have 3.2.2

-The GTK version does not have this small square window in the upper left hand corner
-The QT5 version DOES

So this might not be Lazarus / Lazarus + DE causing this issue?
Could this be QT related?
Or if it is Lazarus / Lazarus + DE related then it could be these in combination with QT?

Not sure if anyone can confirm anything about this?

Thank You

**Reply by W4RH4WK** [reply]
To add a bit of information here: I've confirmed that this does happen with the Qt6 version packaged in Arch, but not with the Qt5 version.
I am running this on my laptop with an NVMe drive, and it is certainly noticeable.
The issue does not appear when doublecmd is launched the first time after boot, but on subsequent launches.

For users with a tiling window manager this is very noticeable as windows get rearranged twice before the real doublecmd window appears.
See video attached to #2425.

**Comment by advice2020** [comment]
@W4RH4WK
Thank you for sharing this information.
Okay so drive type has no connection to this issue, I thought faster drives would have made this less noticeable.
So as it has been mentioned, it is some kind of connection between Lazarus and DE.

Some of your other results though differ from ones I have noticed, wonder why, maybe something Arch related?
For me this does happen in QT5 and QT6 (had to stop using for other reasons).  Also for me this does happen at first launch.

For me
-Every launch = yes
-Properties dialog = yes
-Find tool dialog = yes (really noticeable)

-New folder dialog = no
-Lua script dialogs = no

I know it is just a graphical issue, but to me it can be a little distracting/undesired so if something like this could be fixed that would be great..
I mentioned this before, does anyone know if Lazarus is aware of this is issue, hopefully they know about this?
By any chance does anyone have a Lazarus bug report account that they could mention this or maybe can just link this thread post for Lazarus developers to read?

Thank you to anyone who reads this


W4RH4WK or anyone other MPV users
Sorry, one more thing, I mentioned in a reply above that I also experience this in application called MPV.  Not sure if you use this program, but if you or any others do, would you be willing to try this out as well.
I speak more about this in my reply above, but basically kind of similar thing happens when launch MPV in fullscreen.

Thank you to anyone who reads/confirms

**Reply by W4RH4WK** [reply]
So far I haven't noticed this issue with any other application. I am using MPV (I assume you are referring to the media player), which works fine for me.

I have uploaded some of my [doublecmd config files](https://github.com/W4RH4WK/dotFiles/tree/main/arch/dot-config/doublecmd), maybe there's something relevant in there.

**Reply by j2969719** [reply]
you can probably also encounter this with lazarus-qtX itself for example. what window manager/compositor are you using?

**Reply by W4RH4WK** [reply]
I am using [River WM](https://github.com/riverwm/river), but I'd suspect people using Sway run into this as well.

**Comment by advice2020** [comment]
Did not realize replies for this.

@W4RH4WK
Thank you for response.
Okay so experience this issue with DC QT6 but not with MPV media player.  So maybe these two issues are not related even though they generate the similar results.

Alex mentioned this might be Lazarus bug and also depending on DE.
As mentioned these small windows do not occur with DC GTK2 AppImage so this must Lazarus + DE used + QT based as well.

If any other users are willing to test this out with
DC QT5 / DC QT6
or with
MPV 0.38 / 0.39 / 0.40
please let me know your results

Thank You

Edit:
Does anyone know if Lazarus is aware of this issue?
If not would anyone be willing to report this issue?
Thank You

**Comment by W4RH4WK** [comment]
One more piece of information: I've recently switched to kvantum and this bug is no longer present here, despite using the Qt6 version packaged in Arch.

(using `QT_STYLE_OVERRIDE=kvantum`)

**Comment by advice2020** [comment]
Thank you for response.
Not sure if the information that was shared in the reply above / Lazarus update / some other DC related fix?
but this now appears to be fixed!
Really happy to see that this has been corrected.
I had thought that this might be due specifically to window manager I was using because I had also experienced this with application MPV (which I unfortunately still do) but I guess the two issues were unrelated.

Thank you to anyone involved in this issue for getting this resolved.

Note:
Not sure if related, but I did noticed that DC window button now flashes in Panel at launch?  Not sure if this was in any way due to the fixing of this issue?
I will make an issue post after this just in case.

Double Commander
Version: 1.3.0 alpha
Revision: 13457
Commit: 6072f4fb3
Build: 2026/09/12
Lazarus: 4.99.0.0
Free Pascal: 3.2.2
Platform: x86_64-Linux-qt6
System: Debian 13 based
Desktop Environment: XFCE (x11)
Widgetset library: Qt 6.8.2
